enumerable.rb

```diff
@@ -0,0 +1,159 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    arr = [Hash, Range].member?(self.class) ? to_a.flatten : self
+    i = 0
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    arr
+  end
```

Comment:
- [ ] My each should, when a block is given and self is a range returns the range itself after calling the given block once for each element in self. You are returning an array in this scenario we need to return a range.

- [ ] Same bad return for when using hashes. When a block is given when self is a hash behaves like #each calls the given block once for each element in self. You need to return a hash you are giving a string back.